File: core/src/net/brenig/pixelescape/render/background/AbstractBackgroundLayer.java
package net.brenig.pixelescape.render.background;

import net.brenig.pixelescape.render.WorldRenderer;

/**
 *
 */
public abstract class AbstractBackgroundLayer implements IBackgroundLayer {
	@Override
	public float getBackgroundWorldStart(WorldRenderer renderer, float ratio, int textureSize) {
		return renderer.getWorldCameraXPos() + ((renderer.getWorldCameraXPos() * ratio) % textureSize);
	}

	@Override
	public void onResize(WorldRenderer renderer) {

	}
}

File: core/src/net/brenig/pixelescape/render/background/AbstractBackgroundLayer.kt
package net.brenig.pixelescape.render.background

import net.brenig.pixelescape.render.WorldRenderer

/**
 *
 */
abstract class AbstractBackgroundLayer : IBackgroundLayer {
    override fun getBackgroundWorldStart(renderer: WorldRenderer, ratio: Float, textureSize: Int): Float {
        return renderer.worldCameraXPos + renderer.worldCameraXPos * ratio % textureSize
    }

    override fun onResize(renderer: WorldRenderer) {

    }
}
